scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -653,23 +658,27 @@ protected boolean createExecution(NodeSet nodeSet, Node node, InternalJob job, I
 
                     tlogger.debug(task.getId(), "deploying");
 
-                    String terminateNotificationNodeURL = PAActiveObject.getActiveObjectNode(terminateNotification)
-                                                                        .getNodeInformation()
-                                                                        .getURL();
-
-                    threadPool.submitWithTimeout(new TimedDoTaskAction(job,
-                                                                       taskDescriptor,
-                                                                       launcher,
-                                                                       schedulingService,
-                                                                       terminateNotification,
-                                                                       corePrivateKey,
-                                                                       terminateNotificationNodeURL),
-                                                 DOTASK_ACTION_TIMEOUT,
-                                                 TimeUnit.MILLISECONDS);
-
-                    finalizeStarting(job, task, node, launcher);
-
-                    return true;
+                    try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)